command/upload.go

```diff
@@ -99,12 +115,172 @@ func (cmd *Upload) DoRun(args []string) (err error) {
 		HandleError(err, cmd.opts.VerboseOutput)
 	}
 
-	err = client.UploadDir(dataPath, ds.Root, &stdoutkw{}, 64)
+	size, err := totalTarSize(dataPath)
 	if err != nil {
 		HandleError(err, cmd.opts.VerboseOutput)
 	}
 
-	logrus.Infof("Uploaded data to dataset with ID: %v", ds.DatasetID)
+	header := &data.MetadataHeader{
+		Size:    size,
+		Created: time.Now(),
+		Updated: time.Now(),
+	}
+	metadata := data.NewMetadata(header, data.NewBufferedKeyReadWiter())
+	doneCh := make(chan error)
+	progressCh := make(chan int64)
+	progressBarDoneCh := make(chan struct{})
+	pr, pw := io.Pipe()
+
+	if !cmd.opts.JSONOutput {
+		go ProgressBar(size, progressCh, progressBarDoneCh)
+	} else {
+		go func() {
+			for _ = range progressCh {
+			}
+			progressBarDoneCh <- struct{}{}
+		}()
+	}
+	go func() {
+		defer close(progressCh)
+		err := client.ChunkedUpload(pr, metadata, UploadConcurrency, ds.Root, progressCh)
+		pr.Close()
+		doneCh <- err
+	}()
+
+	err = tardir(dataPath, pw)
+	if err != nil && errors.Cause(err) != io.ErrClosedPipe {
+		HandleError(errors.Wrapf(err, "failed to tar '%s'", dataPath), cmd.opts.VerboseOutput)
+	}
+
+	pw.Close()
+	err = <-doneCh
+	<-progressBarDoneCh
+	if err != nil {
+		HandleError(errors.Wrapf(err, "failed to upload '%s'", dataPath), cmd.opts.VerboseOutput)
+	}
+
```

Comment:
Would it be possible to report the number of deduplicated bytes? the user might be confused on why it was so quick?